.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Ubuntu Benchmark"
+
+on:
+  pull_request:
+    paths:
+      - 'velox/**'
+      - 'third_party/**'
+      - '.github/workflows/benchmark.yml'
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+#TODO  concurrency groups?
+jobs:
+  benchmark:
+    runs-on: 8-core
+    container: ghcr.io/facebookincubator/velox-dev:amd64-ubuntu-22.04-avx
```

Comment:
does this guarantee it will run on a dedicated VM, or this will run on a shared env?

Reply:
The 8-core runner should be equivalent to cci machine-executor, the docker image just replaces the runner base image that is used otherwise and should not affect performance. But we can check the data once it comes in to see if the stability is acceptable.  If it is not we have some alternatives we could pursue.

Reply:
(I choose to use the image to keep build times low, we can always switch to use bundled deps or install system deps in the workflow)